src/betanc.jl

```diff
@@ -0,0 +1,142 @@
+const errmax = 1e-15
+
+#Compute tail of noncentral Beta distribution
+"""
+	ncbeta_tail(x,a,b,lambda)
+
+Compute tail of the noncentral beta distribution.
+"""
+function ncbeta_tail(x::Float64, a::Float64, b::Float64, lambda::Float64)
+    if x <= 0.0
+        return 0.0
+    elseif x >= 1.0
+        return 1.0
+    end
+    
+    c = 0.5*lambda
+    #Init series
+
+    beta = logabsgamma(a)[1] + logabsgamma(b)[1] - logabsgamma(a+b)[1]
+    temp = beta_inc(a,b,x)[1]
+    gx = exp(a*log(x) + b*log(1.0-x) - beta - log(a))
+    q = exp(-c)
+    xj = 0.0
+    ax = q*temp
+    sumq = 1.0 - q
+    ans = ax
+
+    while true
+        xj += 1.0
+        temp -= gx
+        gx *= x*(a+b+xj-1.0)/(a+xj)
+        q *= c/xj
+        sumq -= q
+        ax = temp*q
+        ans += ax
+
+        #Check convergence
+        errbd = abs((temp-gx)*sumq)
+        if trunc(Int,xj) > 1000 || errbd < 1e-10
+            break
+        end
+    end
+    return ans
+end
+
+#R Chattamvelli, R Shanmugam, Algorithm AS 310: Computing the Non-central Beta Distribution #Function, Applied Statistics, Volume 46, Number 1, 1997, pages 146-156
+
+"""
+	ncbeta(a,b,lambda,x)
+
+Compute the CDF of the noncentral beta distribution.
+"""
+function ncbeta(a::Float64, b::Float64, lambda::Float64, x::Float64)
+    ans = x
+    if x <= 0.0
+        return 0.0
+    elseif x >= 1.0
+        return 1.0
+    end
+    c = 0.5*lambda
+    xj = 0.0
+
+    if lambda < 54.0
+        return ncbeta_tail(x,a,b,lambda)
+    else

```

Comment:
split this branch into a separate sub-function, and document algorithm/series.

Reply:
You mean keeping the `else` part in separate function right? What should I name  it?